SPIFlash.cpp

```diff
@@ -115,8 +120,18 @@ uint16_t SPIFlash::readDeviceId()
   select();
   SPI.transfer(SPIFLASH_IDREAD);
 #endif
-  uint16_t jedecid = SPI.transfer(0) << 8;
+  uint32_t jedecid = SPI.transfer(0) << 16;
+  jedecid |= SPI.transfer(0) << 8;
   jedecid |= SPI.transfer(0);
+  if (jedecid == 0x7F7F) { // The 0x7F7F signature means it is a Cypress/Ramtron chip, the real ID will follow in additionnal bytes
```

Comment:
Unfortunately, the whole Cypress/Ramtron is using the 7F7F as a "continuation" ID becaud the returned ID is in fact 72 bits instead 24 bits like most other chips.